tests/services/validation.test.js

```diff
@@ -0,0 +1,31 @@
+const isValidUrl = require("../../services/validation");
+
+describe("isValidUrl", () => {
+  // Test that a URL missing a protocol returns undefined
+  it("returns undefined when the URL is missing a protocol", () => {
+    const url = "www.offerzen.com";
+    const result = isValidUrl(url);
+    expect(result).toBeUndefined();
+  });
+
+  // Test that a URL with a path is returned as is
+  it("returns the URL when it has a path", () => {
```

Comment:
The code to check if the returned URL is the same as the input is duplicated. This can be refactored into a helper function to reduce code duplication

Here is how the helper function should look like
```
const checkUrl = (result, url) => {
  expect(result).toBe(url);
};
```

And the refactored test should look like this 

```
it("should return the URL when it has a path", () => {
    const url = "http://www.a24group.com/path";
    const result = isValidUrl(url);
    checkUrl(result, url);
  });
```